packages/app-page-builder/src/editor/plugins/elementSettings/clone/CloneAction.ts

```diff
@@ -1,11 +1,23 @@
 import React from "react";
+import { cloneDeep } from "lodash";
 import { useEventActionHandler } from "../../../hooks/useEventActionHandler";
 import { activeElementAtom, elementByIdSelector } from "../../../recoil/modules";
 import { plugins } from "@webiny/plugins";
 import { PbEditorPageElementPlugin, PbEditorElement } from "../../../../types";
 import { useRecoilValue } from "recoil";
 import { CloneElementActionEvent } from "../../../recoil/actions/cloneElement";
 
+const removeVarRef = (element: PbEditorElement) => {
+    if (element?.data?.varRef) {
+        const elementCopy = cloneDeep(element);
+        delete elementCopy.data.varRef;
+
+        return elementCopy;
+    } else {
+        return element;
+    }
```

Comment:
Nitpicking here, but you don't need the `else` part, since you're returning within the `if`.

Reply:
inside if func is returning the **cloned** element and in the else - original element is returned
it is not required to clone the element if it is not using a variable (it is pointless, just more cpu work)

The only reason I am cloning the element is to remove varRef (aka id) without affecting the element we are cloning now.
It is a case for element with a variable.
If element is without variables - original element can be returned

Reply:
@neatbyte-vnobis I didn't mean the logic, but the actual `else` statement. So just do the return like this:

```ts
if (...) {
    // your logic
    return whatever;
}

return element;
```

Reply:
got it

Reply:
fixed